src/lib/eliom_client.client.mli

```diff
@@ -245,6 +245,22 @@ type changepage_event =
     apply to the next page change. *)
 val onchangepage : (changepage_event -> unit Lwt.t) -> unit
 
+(* Run some code when arriving at a page. More specifically, this is currently
+   the only way to execute something when a page is being served from the DOM
+   cache upon navigating to a page in the browser history. If [now] is [true]
+   (default) the code is also executed right away.
+   This function is to be used to launch processes that are to run while a page
+   is being viewed and are not attached to an event listener belonging to a DOM
+   element. Specifically event listeners belonging to [Dom_html.window] are
+   destroyed when leaving the page and not reinstated automatically if the page
+   is served from the DOM cache.
+   Be careful to call this function at the right point in time. For instance
+   when called asynchronously it could easily be executed BEFORE the actual
+   page change! So it would be registered with the previously viewed page.
+   TODO: inform this function on which page change it belongs to, so
+   asynchronous calls are possible. *)
```

Comment:
Can you specify that `onarrive` handlers are called for the current page only? `onarrive` does not install a handler for all pages like some other `on*` do.